File: 99_CS50/pset6/cash/cash.py
# derives the minimum number of coins required to provide correct change.

dollars = -1
coins = 0

# gets required change from user.
while dollars < 0:
    try:
        dollars = float(input("Change owed: "))
    except ValueError:
        continue

# turns dollars in to cents and rounds.
cents = round(dollars * 100)

# calc 25c
while (cents - 25 >= 0):
    coins += 1
    cents -= 25

# calc 10c
while (cents - 10 >= 0):
    coins += 1
    cents -= 10

# calc 5c
while (cents - 5 >= 0):
    coins += 1
    cents -= 5

# calc 1c
while (cents - 1 >= 0):
    coins += 1
    cents -= 1

print(coins)

File: 99_CS50/pset1/cash/cash.c
//get libraries
#include <stdio.h>
#include <cs50.h>
#include <math.h>

//start script
int main(void)
{

    //introduce variables
    float dollars;
    int coins = 0;

    //prompt user for required change, validate for number > 0.
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    //convert dollars to cents and round.
    int cents = round(dollars * 100);

    //check for 25c.
    while (cents - 25 >= 0)
    {
        coins++;
        cents = cents - 25;
    }

    //check for 10c.
    while (cents - 10 >= 0)
    {
        coins++;
        cents = cents - 10;
    }

    //check for 5c.
    while (cents - 5 >= 0)
    {
        coins++;
        cents = cents - 5;
    }

    //check for 1c.
    while (cents - 1 >= 0)
    {
        coins++;
        cents = cents - 1;
    }

    //return number of coins required.
    printf("%i\n", coins);

}